packages/zoe/src/contractFacet/zcfMint.js

```diff
@@ -0,0 +1,182 @@
+            zcfSeat.isOfferSafe(allocationPlusGains),
+            `The allocation after minting gains ${allocationPlusGains} for the zcfSeat was not offer safe`,
+          );
+          // No effects above, apart from incrementBy. Note COMMIT POINT within
+          // reallocateForZCFMint. The following two steps *should* be
+          // committed atomically, but it is not a disaster if they are
+          // not. If we minted only, no one would ever get those
+          // invisibly-minted assets.
+          E(zoeMint).mintAndEscrow(totalToMint);
+          reallocateForZCFMint(zcfSeat, allocationPlusGains);
+          return zcfSeat;
+        },
+        burnLosses: (losses, zcfSeat) => {
+          losses = coerceAmountKeywordRecord(losses, getAssetKindByBrand);
+          const totalToBurn = Object.values(losses).reduce(add, empty);
+          assert(
+            !zcfSeat.hasExited(),
+            `zcfSeat must be active to burn losses from the zcfSeat`,
+          );
+          const allocationMinusLosses = subtractFromAllocation(
+            zcfSeat.getCurrentAllocation(),
+            losses,
+          );
+
+          // verifies offer safety
+          assert(
+            zcfSeat.isOfferSafe(allocationMinusLosses),
+            `The allocation after burning losses ${allocationMinusLosses} for the zcfSeat was not offer safe`,
+          );
+
+          // Decrement the stagedAllocation if it exists so that the
+          // stagedAllocation is kept up to the currentAllocation
+          if (zcfSeat.hasStagedAllocation()) {
+            zcfSeat.decrementBy(losses);
+          }
+
+          // No effects above, apart from decrementBy. Note COMMIT POINT within
+          // reallocateForZCFMint. The following two steps *should* be
+          // committed atomically, but it is not a disaster if they are
+          // not. If we only commit the allocationMinusLosses no one would
+          // ever get the unburned assets.
+          reallocateForZCFMint(zcfSeat, allocationMinusLosses);
+          E(zoeMint).withdrawAndBurn(totalToBurn);
+        },
+      },
+    );
+  };
+
+  const makeDurableZcfMint = async (keyword, zoeMint, zcfMintBaggage) => {
+    zcfMintBaggage.init('keyword', keyword);
+    zcfMintBaggage.init('zoeMint', zoeMint);
+    return provideDurableZcfMint(zcfMintBaggage);
+  };
+
+  /**
+   * zcfMintFactory has a method makeZCFMintInternal() that takes a keyword and the
+   * promise returned by a makeZoeMint() call. makeZCFMintInternal() creates a new
+   * baggage for the state of the zcfMint, makes a durableZcfMint from that
+   * baggage, and registers that baggage to be revived with the factory.
+   */
+  const zcfMintFactory = ProvideFar(zcfBaggage, 'zcfMintFactory', {
+    makeZCFMintInternal: async (keyword, zoeMint) => {
+      const zcfMintBaggage = makeScalarBigMapStore('zcfMintBaggage', {
+        durable: true,
+      });
+      // TODO is this AWAIT ok? Or does this need atomicity?
```

Comment:
Are we talking about "atomicity" in the sense of it all happens or none (rollback) or in the sense that it needs to happen in one turn? If the latter it would help to have a distinct word. Maybe "intraturn"?

Wouldn't synchronously returning promise chain instead of await guarantee that?
```
return makeDurableZcfMint(…)
  .then(zcfMint => {
    zcfMintBaggageSet.add(zcfMint);
    return zcfMint;
  });
```

This code comment is throughout the PR but I think they all will have the same answer.

Reply:
There are three of these altogether. The other two are in zcfZygote. @Chris-Hibbert , let's try to find the time to talk all three of these through.

Reply:
> where should this explanation go?

[No Nested Await](https://github.com/Agoric/agoric-sdk/wiki/No-Nested-Await), part of [Coding Style](https://github.com/Agoric/agoric-sdk/wiki/Coding-Style)

Reply:
Thanks. https://github.com/Agoric/agoric-sdk/wiki/%22Atomic%22-vs-%22Transactional%22-terminology-note

Reply:
@erights and I discussed the original issue here and decided there's no problem.